lib/pengine/unpack.c

```diff
@@ -470,9 +470,11 @@ pe_create_node(const char *id, const char *uname, const char *type,
     }
 
     new_node->weight = char2score(score);
-    new_node->details = calloc(1, sizeof(struct pe_node_shared_s));
 
-    if (new_node->details == NULL) {
+    new_node->details = calloc(1, sizeof(struct pe_node_shared_s));
+    new_node->private = calloc(1, sizeof(pcmk__node_private_t));
+    if ((new_node->details == NULL) || (new_node->private == NULL)) {
+        free(new_node->details);
```

Comment:
If you want to be especially thorough, free both. It should be theoretically possible that the first `calloc()` fails and the second succeeds. In that case, `details` is `NULL` and `private` is not.

Would be shocking if that happened in practice, but I wouldn't be surprised if coverity complains now or in the future.

Reply:
true that